# Quality control of Sequencing Reads

## Potential QC tools
- Seqkit
- Fastqc
- MultiQC
- BBDuk script
- Trimmomatic
- Kneaddata

## Required resources
- Adapter fasta files
- PhiX fasta file

## QC environment.yml

```bash
name: readqc
channels:
    - bioconda
    - biobakery
dependencies:
    - seqkit =2.3.1
    - fastqc =0.12.1
    - multiqc =1.14
    - bbmap =39.01
    - trimmomatic =0.39
    - knead-data =0.12.0
```

```bash
conda activate base
mamba install -c bioconda -c conda-forge -n readqc -file environment.yml
```


- We assume that the tools above are installed. Now it is time to use them to:
  - Get simple statistics.
  - Check the quality of the reads.
  - Create summary report of quality metrics.
  - Trim poor read at a user-specified cutoff.
  - Remove contaminants.

<br><br>

## Raw reads
![](results/qc/multiqc1/multiqc_plots/svg/mqc_fastqc_per_base_sequence_quality_plot_1.svg)

<br><br>

## After trimming poor reads
![](results/qc/multiqc2/multiqc_plots/svg/mqc_fastqc_per_base_sequence_quality_plot_1.svg)

<br><br>

## After removing contaminated reads
![](results/qc/multiqc2/multiqc_plots/svg/mqc_fastqc_per_base_sequence_quality_plot_1.svg)

<br><br>

## Graphical view of processed read status

![](images/qc_hist.svg)


## Snakemake workflow rules

<br>

![Tentative snakemake workflow](dags/rulegraph.svg)

<br>